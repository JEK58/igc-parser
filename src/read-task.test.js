const readTask = require('./read-task');

describe('readTask()', () => {
  it('reads "2017-07-15-lev" task correctly', () => {
    let task = readTask(`${__dirname}/../fixtures/2017-07-15-lev/task.tsk`);
    expect(task).toMatchSnapshot();
  })
});
