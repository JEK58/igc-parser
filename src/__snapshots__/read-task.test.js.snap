// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readTask() reads "2017-07-15-lev" task correctly 1`] = `
Object {
  "aatMinTime": 10800,
  "points": Array [
    Object {
      "observationZone": Line {
        "center": Array [
          6.9852833,
          51.1413833,
        ],
        "length": 20000,
      },
      "type": "Start",
      "waypoint": Object {
        "altitude": 86,
        "location": Array [
          6.9852833,
          51.1413833,
        ],
        "name": "Langenfeld-Wiescheid",
      },
    },
    Object {
      "observationZone": Cylinder {
        "center": Array [
          6.18695,
          50.8241667,
        ],
        "radius": 10000,
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 190,
        "location": Array [
          6.18695,
          50.8241667,
        ],
        "name": "Aachen Merzbrück",
      },
    },
    Object {
      "observationZone": Cylinder {
        "center": Array [
          7.1375,
          50.5597167,
        ],
        "radius": 30000,
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 205,
        "location": Array [
          7.1375,
          50.5597167,
        ],
        "name": "Bad Neuenahr",
      },
    },
    Object {
      "observationZone": Cylinder {
        "center": Array [
          6.4916667,
          50.8888833,
        ],
        "radius": 10000,
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 7,
        "location": Array [
          6.4916667,
          50.8888833,
        ],
        "name": "058Hambach Süd",
      },
    },
    Object {
      "observationZone": Cylinder {
        "center": Array [
          7.2,
          51.17695,
        ],
        "radius": 10000,
      },
      "type": "Area",
      "waypoint": Object {
        "altitude": 200,
        "location": Array [
          7.2,
          51.17695,
        ],
        "name": "110Remscheid Bhf",
      },
    },
    Object {
      "observationZone": Line {
        "center": Array [
          7.03695,
          51.0977833,
        ],
        "length": 1000,
      },
      "type": "Finish",
      "waypoint": Object {
        "altitude": 130,
        "location": Array [
          7.03695,
          51.0977833,
        ],
        "name": "002Zielkreis",
      },
    },
  ],
  "type": "AAT",
}
`;
